### Importamos las librerias

In [3]:
import pandas as pd
import numpy as np
import random as rd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

%matplotlib inline
plt.style.use('default')
plt.rcParams['figure.figsize'] = (20, 10)
sns.set(style="whitegrid")
pd.options.display.float_format = '{:20,.2f}'.format
warnings.filterwarnings('ignore')

### Importamos los datasets

In [6]:
df_train_values = pd.read_csv('../../datasets/train_values.csv')
df_train_labels = pd.read_csv('../../datasets/train_labels.csv')

df_train_values=df_train_values.astype({
    'land_surface_condition':'category',
    'foundation_type':'category',
    'roof_type':'category',
    'ground_floor_type':'category',
    'other_floor_type':'category',
    'position':'category',
    'plan_configuration':'category',

    'has_superstructure_adobe_mud':'bool',
    'has_superstructure_mud_mortar_stone':'bool',
    'has_superstructure_stone_flag':'bool',
    'has_superstructure_cement_mortar_stone':'bool', 
    'has_superstructure_mud_mortar_brick':'bool', 
    'has_superstructure_cement_mortar_brick':'bool', 
    'has_superstructure_timber':'bool', 
    'has_superstructure_bamboo':'bool', 
    'has_superstructure_rc_non_engineered':'bool', 
    'has_superstructure_rc_engineered':'bool', 
    'has_superstructure_other':'bool',

    'legal_ownership_status':'category',

    'has_secondary_use':'bool',
    'has_secondary_use_agriculture':'bool', 
    'has_secondary_use_hotel':'bool', 
    'has_secondary_use_rental':'bool', 
    'has_secondary_use_institution':'bool', 
    'has_secondary_use_school':'bool',
    'has_secondary_use_industry':'bool',
    'has_secondary_use_health_post':'bool',
    'has_secondary_use_gov_office':'bool',
    'has_secondary_use_use_police':'bool',
    'has_secondary_use_other':'bool'
})
df_train_labels = df_train_labels.astype({'damage_grade':'category'})

### Hacemos el análisis

In [5]:
df_has_secondary = df_train_values[[
    'has_secondary_use',
    'has_secondary_use_agriculture', 
    'has_secondary_use_hotel', 
    'has_secondary_use_rental', 
    'has_secondary_use_institution', 
    'has_secondary_use_school',
    'has_secondary_use_industry',
    'has_secondary_use_health_post',
    'has_secondary_use_gov_office',
    'has_secondary_use_use_police',
    'has_secondary_use_other'
]]

In [7]:
df_has_secondary.head()

has_secondary_use  has_secondary_use_agriculture  has_secondary_use_hotel  \
0              False                          False                    False   
1              False                          False                    False   
2              False                          False                    False   
3              False                          False                    False   
4              False                          False                    False   

   has_secondary_use_rental  has_secondary_use_institution  \
0                     False                          False   
1                     False                          False   
2                     False                          False   
3                     False                          False   
4                     False                          False   

   has_secondary_use_school  has_secondary_use_industry  \
0                     False                       False   
1                     False                       False   
2                     False                       False   
3                     False                       False   
4                     False                       False   

   has_secondary_use_health_post  has_secondary_use_gov_office  \
0                          False                         False   
1                          False                         False   
2                          False                         False   
3                          False                         False   
4                          False                         False   

   has_secondary_use_use_police  has_secondary_use_other  
0                         False                    False  
1                         False                    False  
2                         False                    False  
3                         False                    False  
4                         False                    False

In [8]:
df_has_secondary.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 260601 entries, 0 to 260600
Data columns (total 11 columns):
 #   Column                         Non-Null Count   Dtype
---  ------                         --------------   -----
 0   has_secondary_use              260601 non-null  bool 
 1   has_secondary_use_agriculture  260601 non-null  bool 
 2   has_secondary_use_hotel        260601 non-null  bool 
 3   has_secondary_use_rental       260601 non-null  bool 
 4   has_secondary_use_institution  260601 non-null  bool 
 5   has_secondary_use_school       260601 non-null  bool 
 6   has_secondary_use_industry     260601 non-null  bool 
 7   has_secondary_use_health_post  260601 non-null  bool 
 8   has_secondary_use_gov_office   260601 non-null  bool 
 9   has_secondary_use_use_police   260601 non-null  bool 
 10  has_secondary_use_other        260601 non-null  bool 
dtypes: bool(11)
memory usage: 2.7 MB


In [10]:
df_has_secondary_grouped = df_has_secondary.groupby(['has_secondary_use']).sum().reset_index()

In [11]:
df_has_secondary_grouped

has_secondary_use  has_secondary_use_agriculture  has_secondary_use_hotel  \
0              False                              0                        0   
1               True                          16777                     8763   

   has_secondary_use_rental  has_secondary_use_institution  \
0                         0                              0   
1                      2111                            245   

   has_secondary_use_school  has_secondary_use_industry  \
0                         0                           0   
1                        94                         279   

   has_secondary_use_health_post  has_secondary_use_gov_office  \
0                              0                             0   
1                             49                            38   

   has_secondary_use_use_police  has_secondary_use_other  
0                             0                        0  
1                            23                     1334

In [12]:
df_has_secondary_grouped.sum(axis=1)

0                   0.00
1              29,714.00
dtype: float64

No hay inconsistencias de los datos

In [13]:
df_has_secondary_true = df_has_secondary.loc[df_has_secondary['has_secondary_use']==1,:]

In [14]:
df_has_secondary_true.head()

has_secondary_use  has_secondary_use_agriculture  has_secondary_use_hotel  \
5                True                           True                    False   
20               True                           True                    False   
26               True                           True                    False   
45               True                          False                    False   
57               True                           True                    False   

    has_secondary_use_rental  has_secondary_use_institution  \
5                      False                          False   
20                     False                          False   
26                     False                          False   
45                      True                          False   
57                     False                          False   

    has_secondary_use_school  has_secondary_use_industry  \
5                      False                       False   
20                     False                       False   
26                     False                       False   
45                     False                       False   
57                     False                       False   

    has_secondary_use_health_post  has_secondary_use_gov_office  \
5                           False                         False   
20                          False                         False   
26                          False                         False   
45                          False                         False   
57                          False                         False   

    has_secondary_use_use_police  has_secondary_use_other  
5                          False                    False  
20                         False                    False  
26                         False                    False  
45                         False                    False  
57                         False                    False

In [15]:
del(df_has_secondary_true['has_secondary_use'])

In [17]:
df_has_secondary_true['Suma_secondary'] = df_has_secondary_true.sum(axis=1)
df_has_secondary_true['Suma_secondary'].value_counts()

2.00    28599
4.00      557
Name: Suma_secondary, dtype: int64

In [18]:
df_has_secondary_true['Suma_secondary'].value_counts().sum()

29156

29156 edificios tiene uso secundario y 557 tienen más de un tipo de uso.

In [21]:
df_has_secondary['Suma_secondary'] = df_has_secondary.sum(axis=1)
df_has_mas_secondary = df_train_values[(df_has_secondary['Suma_secondary'] > 2)]
df_has_mas_secondary.iloc[:,29:]

has_secondary_use_agriculture  has_secondary_use_hotel  \
5                                True                    False   
20                               True                    False   
26                               True                    False   
45                              False                    False   
57                               True                    False   
...                               ...                      ...   
260579                           True                    False   
260584                           True                    False   
260587                           True                    False   
260591                           True                    False   
260593                           True                    False   

        has_secondary_use_rental  has_secondary_use_institution  \
5                          False                          False   
20                         False                          False   
26                         False                          False   
45                          True                          False   
57                         False                          False   
...                          ...                            ...   
260579                     False                          False   
260584                     False                          False   
260587                     False                          False   
260591                     False                          False   
260593                     False                          False   

        has_secondary_use_school  has_secondary_use_industry  \
5                          False                       False   
20                         False                       False   
26                         False                       False   
45                         False                       False   
57                         False                       False   
...                          ...                         ...   
260579                     False                       False   
260584                     False                       False   
260587                     False                       False   
260591                     False                       False   
260593                     False                       False   

        has_secondary_use_health_post  has_secondary_use_gov_office  \
5                               False                         False   
20                              False                         False   
26                              False                         False   
45                              False                         False   
57                              False                         False   
...                               ...                           ...   
260579                          False                         False   
260584                          False                         False   
260587                          False                         False   
260591                          False                         False   
260593                          False                         False   

        has_secondary_use_use_police  has_secondary_use_other  
5                              False                    False  
20                             False                    False  
26                             False                    False  
45                             False                    False  
57                             False                    False  
...                              ...                      ...  
260579                         False                    False  
260584                         False                    False  
260587                         False                    False  
260591                         False                    False  
260593                         False                    False  

[29156 rows x 10 co

In [22]:
df_has_mas_secondary['has_secondary_use_other'].value_counts()

False    27822
True      1334
Name: has_secondary_use_other, dtype: int64

Los 557 casos de dos valores true en secondary use tiene uno de esos en la columna has_secondary_use